velox/type/Timestamp.cpp

```diff
@@ -249,88 +234,159 @@ bool Timestamp::epochToUtc(int64_t epoch, std::tm& tm) {
   return true;
 }
 
-std::string Timestamp::tmToString(
+StringView Timestamp::tmToStringView(
     const std::tm& tmValue,
     uint64_t nanos,
-    const TimestampToStringOptions& options) {
+    const TimestampToStringOptions& options,
+    char* const startPosition) {
   VELOX_DCHECK_GE(nanos, 0);
   VELOX_DCHECK_LT(nanos, 1'000'000'000);
-  auto precisionWidth = static_cast<int8_t>(options.precision);
-  std::string out;
-  out.reserve(getCapacity(options));
 
+  const auto appendDigits = [](const int value,
+                               const std::optional<uint32_t> minWidth,
+                               char* const position) {
+    const auto numDigits = countDigits(value);
+    uint32_t offset = 0;
+    // Append leading zeros when there is the requirement for minumum width.
+    if (minWidth.has_value() && numDigits < minWidth.value()) {
+      const auto leadingZeros = minWidth.value() - numDigits;
+      std::memset(position, '0', leadingZeros);
+      offset += leadingZeros;
+    }
+    const auto [endPosition, errorCode] =
+        std::to_chars(position + offset, position + offset + numDigits, value);
+    VELOX_DCHECK_EQ(
```

Comment:
Would be nice to use Status to report errors.

Reply:
Thanks for the comment. Here we do not expect an error because the input is a valid `Timestamp` value. We suppose the conversion must succeed, otherwise a bug may exist in the code. Do you it makes sense?

Reply:
Got it. Thank you for clarifying.